pennylane/templates/subroutines/phase_adder.py

```diff
@@ -216,12 +221,20 @@ def compute_decomposition(k, x_wires, mod, work_wire):  # pylint: disable=argume
         else:
             aux_k = x_wires[0]
             op_list.extend(_add_k_fourier(k, x_wires))
-            op_list.extend(qml.adjoint(_add_k_fourier)(mod, x_wires))
+
+            # Adjoint is not iterable in QJIT
+            for op in reversed(_add_k_fourier(mod, x_wires)):
+                op_list.append(qml.adjoint(op))
+
             op_list.append(qml.adjoint(qml.QFT)(wires=x_wires))
             op_list.append(qml.ctrl(qml.PauliX(work_wire), control=aux_k, control_values=1))
             op_list.append(qml.QFT(wires=x_wires))
             op_list.extend(qml.ctrl(op, control=work_wire) for op in _add_k_fourier(mod, x_wires))
-            op_list.extend(qml.adjoint(_add_k_fourier)(k, x_wires))
+
+            # Adjoint is not iterable in QJIT
```

Comment:
bu why is `adjoint` iterable normally 🤔

Reply:
When adjoint is called on a Callable like this it will call `make_qscript` then return a list `[Adjoint(op) for op in reversed(qscript.operations)]`.